src/ResourceManager/IotHub/Commands.IotHub/IotHub/AddAzureRmIotHubCertificate.cs

```diff
@@ -0,0 +1,172 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System;
+    using System.Globalization;
+    using System.IO;
+    using System.Management.Automation;
+    using System.Text;
+    using Microsoft.Azure.Commands.Management.IotHub.Common;
+    using Microsoft.Azure.Commands.Management.IotHub.Models;
+    using Microsoft.Azure.Management.IotHub;
+    using Microsoft.Azure.Management.IotHub.Models;
+    using ResourceManager.Common.ArgumentCompleters;
+
+    [Cmdlet(VerbsCommon.Add, "AzureRmIotHubCertificate", DefaultParameterSetName = ResourceParameterSet, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSCertificateDescription))]
+    public class AddAzureRmIotHubCertificate : IotHubBaseCmdlet
+    {
+        private const string ResourceIdParameterSet = "ResourceIdSet";
+        private const string ResourceParameterSet = "ResourceSet";
+        private const string InputObjectParameterSet = "InputObjectSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Certificate Object")]
+        [ValidateNotNullOrEmpty]
+        public PSCertificateDescription InputObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the Resource Group")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the Iot Hub")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            HelpMessage = "Name of the Certificate")]
+        [Parameter(
```

Comment:
For add-certificate, there will be no certificate object, so the inputobject and resourceid should be an IotHub. Please fix the parameters and logic to reflect this.

Reply:
In add-certificate, certificate can create or update so if etag will be mention then it will work as update certificate and in that scenario inputobject or resourceid will work

Reply:
Okay, that makes sense.